calendar-app/Components/Storage/userDataStorage.js

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+import store from 'react-native-simple-store';
+
+export const useUserSettings = (userIDKey, inGroupKey) => {
+  const [settings, setSettings] = useState({ userID: null, inGroup: null, isLoaded: false });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const [storedUserID, storedInGroup] = await Promise.all([
+          store.get(userIDKey),
+          store.get(inGroupKey),
+        ]);
+
+        setSettings((prevState) => ({
+          ...prevState,
+          userID: storedUserID !== null ? parseFloat(storedUserID) : null,
+          inGroup: storedInGroup !== null ? storedInGroup : null,
+          isLoaded: true,
+        }));
+      } catch (error) {
+        logError('fetching user settings from store', error);
+      }
+    };
+
+    fetchData();
+  }, [userIDKey, inGroupKey]);
+
+  const logError = (action, error) => {
+    console.error(`Error ${action}:`, error);
+  };
+
+  const saveSetting = async (key, value, setter) => {
+    try {
+      await store.save(key, typeof value === 'number' ? value.toString() : value);
```

Comment:
We probably will be storing a UUID Data type, so will this be able to store something like : cba1dd22-5815-4344-b202-914cdc8da27f or "cba1dd22-5815-4344-b202-914cdc8da27f"

Reply:
Yes it should be possible to save the UUID data type as the await store.save(key, typeof value === 'number' ? value.toString() : value); has an option to save string type information which can save the UUID information.